platform/mbed_rtc_time.h

```diff
@@ -63,6 +63,10 @@ extern "C" {
  *
  * @note Synchronization level: Thread safe
  *
+ * @note On some platforms time can not be set to 0 since it is invalid value.
+ *       In such case when 0 value is passed to this function, then RTC time
+ *       value is set to 1.
+ *
```

Comment:
I would consider this a bug with the vendor implementation. If 0 is unsupported by hardware then they should have a software translation.

I would argue code like [this](https://github.com/ARMmbed/mbed-os/blob/master/targets/TARGET_Freescale/TARGET_MCUXpresso_MCUS/api/rtc_api.c#L53-L66) should be updated to have an offset instead:
```c
#define RTC_OFFSET 1
time_t rtc_read(void)
{
    return (time_t)(RTC->TSR - RTC_OFFSET);
}

void rtc_write(time_t t)
{
    RTC_StopTimer(RTC);
    RTC->TSR = t + RTC_OFFSET;
    RTC_StartTimer(RTC);
}
```